transaction.go

```diff
@@ -164,20 +167,29 @@ func (t *Transaction) commit() error {
 
 // marshalUpdates marshals the updates of a transaction
 func marshalUpdates(updates []Update) []byte {
+	// Declare helper function to find minimum
+	min := func(a, b int) int {
+		if a < b {
+			return a
+		}
+		return b
+	}
+
 	// preallocate buffer of appropriate size
 	var size int
 	for _, u := range updates {
-		size += 8 + len(u.Name)
+		size += 1 + min(len(u.Name), math.MaxUint8)
```

Comment:
this raises a good question, which is how we want to deal with update names that are longer than 256 bytes. I think silently truncating them is the wrong approach. During recovery, the caller will assume that they can write code like:
```go
for _, u := range recoveredUpdates {
    switch u.Name {
    case myReallyLongUpdateName1:
        // ...
    case myReallyLongUpdateName2:
        // ...
    }
}
```
but if we silently truncate, the `switch` won't match any of the non-truncated names.

Instead, we probably have to enforce the length restriction in `NewTransaction` and `Append`. We can also use this to ban names with length 0, as required by #60. I think the only question there is: should we return an error, or panic? If your update is invalid, it's always going to be invalid, and that to me signals a developer error, for which a panic might be appropriate.

Reply:
Hm I guess an error would probably be better. I guess it's better to use panics to indicate an error on the developer's side (us) and an error to indicate user errors.